packages/core/src/components/forms/numericInput.tsx

```diff
@@ -256,16 +271,19 @@ export class NumericInput extends AbstractComponent<HTMLInputProps & INumericInp
     }
 
     public componentDidUpdate() {
-        if (this.state.shouldSelectAfterUpdate) {
+        if (this.shouldSelectAfterUpdate) {
             this.inputElement.setSelectionRange(0, this.state.value.length);
         }
     }
 
     protected validateProps(nextProps: HTMLInputProps & INumericInputProps) {
-        const { majorStepSize, max, min, minorStepSize, stepSize } = nextProps;
-        if (min && max && min >= max) {
+        const { clampValueOnBlur, majorStepSize, max, min, minorStepSize, stepSize } = nextProps;
+        if (min != null && max != null && min >= max) {
             throw new Error(Errors.NUMERIC_INPUT_MIN_MAX);
         }
+        if (clampValueOnBlur && min == null && max == null) {
+            throw new Error(Errors.NUMERIC_INPUT_CLAMP_VALUE_ON_BLUR);
```

Comment:
Yeah, this is a little heavy-handed. Maybe could be just a warning to let you know that your clamp flag will have no effect; I'd find that helpful.